angularSTEP/src/app/user-page/user-page.component.ts

```diff
@@ -55,4 +74,62 @@ export class UserPageComponent implements OnInit {
       }
     }
   }
+
+  async isUserFollowing() {
+    if (this.currentPageUid && this.loggedIn && this.currentUserUid) {
+      const postUser = await this.afs.firestore
+        .doc('/users/' + this.currentUserUid + '/')
+        .withConverter(new Converter().userConverter)
+        .get();
+      if (postUser) {
+        this.currentUserData = postUser.data();
+        if (
+          this.currentUserData &&
+          this.currentUserData.following.indexOf(this.currentPageUid) > -1
+        ) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  follow() {
+    if (this.currentUserData && this.currentUserData.following) {
+      this.currentUserData.following.push(this.currentPageUid);
+      this.fAuth.currentUser.then(user => {
+        if (user) {
+          this.afs
+            .collection('users')
+            .doc(this.currentUserUid)
+            .ref.withConverter(new Converter().userConverter)
+            .update({following: this.currentUserData!.following})
+            .then(() => {
+              this.userFollowing = true;
+            });
+        }
+      });
+    }
+  }
+
+  unfollow() {
```

Comment:
Would be a bit cleaner to have one setFollowing(bool) method that had all the storage logic in it, and turn follow() and unfollow() into one-line methods that call setFollowing(true/false).

Reply:
Thanks for the idea, it definitely made the readability better and condensed the code.